src/lib/operators/join_hash.cpp

```diff
@@ -21,54 +21,69 @@
-      _radix_bits(radix_bits) {
-  Assert(primary_predicate.predicate_condition == PredicateCondition::Equals,
-         "Unsupported primary PredicateCondition.");
-  Assert(mode != JoinMode::FullOuter, "Full outer joins are not supported by JoinHash.");
-  Assert(mode != JoinMode::AntiNullAsTrue || _secondary_predicates.empty(),
-         "AntiNullAsTrue joins are not supported by JoinHash with secondary predicates.");
-}
+      _radix_bits(radix_bits) {}
 
 const std::string JoinHash::name() const { return "JoinHash"; }
 
 std::shared_ptr<AbstractOperator> JoinHash::_on_deep_copy(
     const std::shared_ptr<AbstractOperator>& copied_input_left,
     const std::shared_ptr<AbstractOperator>& copied_input_right) const {
-  return std::make_shared<JoinHash>(copied_input_left, copied_input_right, _mode, _primary_predicate, _radix_bits,
-                                    _secondary_predicates);
+  return std::make_shared<JoinHash>(copied_input_left, copied_input_right, _mode, _primary_predicate,
+                                    _secondary_predicates, _radix_bits);
 }
 
 void JoinHash::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
 std::shared_ptr<const Table> JoinHash::_on_execute() {
-  std::shared_ptr<const AbstractOperator> build_operator;
-  std::shared_ptr<const AbstractOperator> probe_operator;
-  ColumnID build_column_id;
-  ColumnID probe_column_id;
-
-  // This is the expected implementation for swapping tables:
-  // (1) if left or right outer join, outer relation becomes probe relation (we have to swap only for left outer)
-  // (2) for a Semi, AntiRetainNull and AntiNullAsTrue the inputs are always swapped
-  // (3) else the smaller relation will become build relation, the larger the probe relation
-  bool inputs_swapped =
+  Assert(supports(_mode, _primary_predicate.predicate_condition,
+                  input_table_left()->column_data_type(_primary_predicate.column_ids.first),
+                  input_table_right()->column_data_type(_primary_predicate.column_ids.second),
+                  !_secondary_predicates.empty()),
+         "JoinHash doesn't support these parameters");
+
+  std::shared_ptr<const AbstractOperator> build_input_operator;
+  std::shared_ptr<const AbstractOperator> probe_input_operator;
+  auto build_column_id = ColumnID{};
+  auto probe_column_id = ColumnID{};
+
+  /**
+   * Build and probe side are assigned as follows (depending only on JoinMode, except for inner joins where input
+   * relation sizes are considered):
+   *
+   * JoinMode::Inner        The smaller relation becomes the build side, the bigger the probe side
+   * JoinMode::Left/Right   The outer relation becomes the probe side, the inner relation becomes the build side
+   * JoinMode::FullOuter    Not supported by JoinHash
+   * JoinMode::Semi/Anti*   The inner relation becomes the build side, the outer relation becomes the probe side
```

Comment:
Is there an inner and outer relation for semi/anti joins?

Reply:
Mmh, they are not called like that, right? reverting to left/right then.